lib/iupac.rb

```diff
@@ -3,22 +3,26 @@
 module Iupac_convention
 
   Suffix =                    #   This is a constant definition
-      {                     
+      {                       # This is a hash (i.e. associative array  where a number of 'keys' are associated with corresponding 'values')
     :alkane => "ane",
      :alcohol => "ol",
      
   }
   
   
```

Comment:
if you want to split a string into an array of strings at each, or at  each
block of, whitespace, bracket or hyphen charactere
"a     b- - () cc[d  ]]--ee".split(/[ \t\r\n\f\-\[\]\(\)]*/)       '
=>["a","b","cc","d","ee"]

On 17 March 2015 at 15:49, mohitgautam05 <notifications@github.com> wrote:

> In lib/iupac.rb
> <https://github.com/nicolejung/IUPAC/pull/2#discussion_r26578829>:
>
> > @@ -3,22 +3,26 @@
> >  module Iupac_convention
> >
> >    Suffix =                    #   This is a constant definition
> > -      {
> > +      {                       # This is a hash (i.e. associative array  where a number of 'keys' are associated with corresponding 'values')
> >      :alkane => "ane",
> >       :alcohol => "ol",
> >
> >    }
> >
> >
>
> If I use "compound_name.split(/ /)" this would separated each word,
> bracket, hyphen and space. How do i then put this into an array of strings
> again ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nicolejung/IUPAC/pull/2/files#r26578829>.
>